kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1168 @@
+
+    typedef ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+
+    typedef typename BaseType::TSchemeType TSchemeType;
+
+    typedef typename BaseType::TDataType TDataType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+
+    typedef typename BaseType::NodeType NodeType;
+
+    typedef typename BaseType::NodesArrayType NodesArrayType;
+    typedef typename BaseType::ElementsArrayType ElementsArrayType;
+    typedef typename BaseType::ConditionsArrayType ConditionsArrayType;
+
+    typedef typename BaseType::ElementsContainerType ElementsContainerType;
+
+    typedef MasterSlaveConstraint MasterSlaveConstraintType;
+    typedef typename MasterSlaveConstraint::Pointer MasterSlaveConstraintPointerType;
+
+    typedef AuxilaryGlobalMasterSlaveRelation AuxilaryGlobalMasterSlaveRelationType;
+
+    typedef PointerVectorSet<AuxilaryGlobalMasterSlaveRelationType, IndexedObject> GlobalMasterSlaveRelationContainerType;
+
+
+/*    typedef PointerVectorSet<
+		            AuxilaryGlobalMasterSlaveRelationType,
+		            IndexedObject,
+		            std::less<typename SetIdentityFunction<AuxilaryGlobalMasterSlaveRelationType>::result_type>,
+		            std::equal_to<typename SetIdentityFunction<AuxilaryGlobalMasterSlaveRelationType>::result_type>,
+		            Kratos::unique_ptr<AuxilaryGlobalMasterSlaveRelationType> > GlobalMasterSlaveRelationContainerType ;*/
+
+    typedef std::vector<IndexType> EquationIdVectorType;
+    typedef std::vector<IndexType> VectorIndexType;
+    typedef std::vector<Dof<double>::Pointer> DofsVectorType;
+    typedef Matrix MatrixType;
+    typedef Vector VectorType;
+    typedef Geometry<NodeType> GeometryType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /** Constructor.
+     */
+    ResidualBasedBlockBuilderAndSolverWithConstraints(
+        typename TLinearSolver::Pointer pNewLinearSystemSolver)
+        : ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>(pNewLinearSystemSolver)
+    {
+    }
+
+    /** Destructor.
+     */
+    ~ResidualBasedBlockBuilderAndSolverWithConstraints() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetUpSystem(
+        ModelPart &rModelPart) override
+    {
+        BaseType::SetUpSystem(rModelPart);
+        if(rModelPart.NumberOfMasterSlaveConstraints() > 0)
+        {
```

Comment:
brackets are not needed here

Reply:
I am in favour of leave the brackets, I feel more safe